api/src/clojure/org/akvo/flow_api/datastore/form_instance.clj

```diff
@@ -103,9 +104,14 @@
   [_ response-str]
   response-str)
 
+(def ^DateFormat sdf (SimpleDateFormat. "dd-MM-yyyy HH:mm:ss z"))
```

Comment:
Good catch!

I think we have two options:

```
(locking sdf
    (.parse sdf response-str))
```

or create a new one every time
```
(let [date-format (SimpleDateFormat. "dd-MM-yyyy HH:mm:ss z")]
  (parse date-format response-str))
```